### PR TITLE
SQL: Fix translation of math functions to painless

### PR DESCRIPTION
**Pull request by matriv**
`SIGN` and `RADIANS` where wrongly overriding `mathFunction()`.
Converted `mathFunction()` to private in `MathFunction` since it
shouldn't be overriden, as it uses the assigned `MathOperation`
to get the funtion name for painless scripts.

Fixes: #35654

### CODE REVIEW

**Review by astefan — APPROVED** [review]
LGTM. I like the test!

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/math/MathFunction.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public String processScript(String template) {
         return super.processScript(format(Locale.ROOT, "{sql}.%s(%s)", mathFunction(), template));
     }
 
-    protected String mathFunction() {
+    private String mathFunction() {
```
Comment (costin) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/math/MathFunction.java]: Let's remove this method all together and simply use the classname inside the template.

**Review by costin — APPROVED** [review]
left a comment.

**Comment by matriv** [comment]
Backported to `6.x` with 969d5608fb04b79b02745d8a28f4f2cbac00a9b3

**Comment by matriv** [comment]
Backported to `6.5` with e4b6eb49fd787dcc2cd7dbdfadf26827cbf677a8

**Pull request merged**